core/commands/log.py

```diff
@@ -9,12 +9,13 @@
 class GsLogCommand(WindowCommand, GitCommand):
     cherry_branch = None
 
-    def run(self, filename=None, limit=6000, author=None, log_current_file=False):
+    def run(self, filename=None, limit=6000, author=None, log_current_file=False, target_hash=None):
         self._pagination = 0
         self._filename = filename
         self._limit = limit
         self._author = author
         self._log_current_file = log_current_file
+        self._target_hash = target_hash
```

Comment:
As you can see  in cherry pick command we use _target_hash if it is in use. 
```python 
class GsCherryPickCommand(GsLogCommand, BranchesMixin):
    def run_async(self):
        if self._target_hash:
            return self.cherry_pick(self._target_hash)
```
This class inherit from GsLogCommand. I think the way it was written was impossible to set self._target_hash.